diff --git a/test/e2e/domainmapping/domain_mapping_test.go b/test/e2e/domainmapping/domain_mapping_test.go
index 9b5c7b2d1..7f71686ce 100644
--- a/test/e2e/domainmapping/domain_mapping_test.go
+++ b/test/e2e/domainmapping/domain_mapping_test.go
@@ -82,7 +82,8 @@ func TestBYOCertificate(t *testing.T) {
 	cert, key := makeCertificateForDomain(t, host)
 	secret, err := clients.KubeClient.CoreV1().Secrets(test.ServingFlags.TestNamespace).Create(ctx, &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name: test.AppendRandomString("byocert-secret"),
+			Name:   test.AppendRandomString("byocert-secret"),
+			Labels: map[string]string{"networking.internal.knative.dev/certificate-uid": "byocert-secret"},
 		},
 		Type: corev1.SecretTypeTLS,
 		Data: map[string][]byte{

diff --git a/openshift/release/knative-serving-ci.yaml b/openshift/release/knative-serving-ci.yaml
index 2531f7579..f5e86b3ff 100644
--- a/openshift/release/knative-serving-ci.yaml
+++ b/openshift/release/knative-serving-ci.yaml
@@ -3516,6 +3516,7 @@ metadata:
   namespace: knative-serving
   labels:
     serving-certs-ctrl: "data-plane"
+    networking.internal.knative.dev/certificate-uid: "data-plane"
 # The data is populated when internal-encryption is enabled.
 ---
 # Copyright 2018 The Knative Authors
 